podcasts/TranscriptModel.swift

```diff
@@ -58,4 +64,39 @@ struct TranscriptModel: Sendable {
     var isEmtpy: Bool {
         return attributedText.string.trim().isEmpty
     }
+
+    private static func extractSpeaker(from cue: Subtitles.Cue, format: TranscriptFormat) -> String? {
+        if let speaker = cue.speaker {
+            return speaker
+        }
+        switch format {
+        case .vtt:
+            return regexMatch(input: cue.text, pattern: "<v (.+?)>", position: 1)
+        case .srt:
+            return regexMatch(input: cue.text, pattern: "^(.+?):", position: 1)
+        default:
+            return nil
+        }
+    }
+
+    private static func regexMatch(input: String, pattern: String, position: Int = 0) -> String? {
```

Comment:
Would it make sense to add an iOS 16 implementation using [Regex](https://developer.apple.com/documentation/swift/regex) or wait to update later?